Aliucord/src/main/java/com/aliucord/coreplugins/NewPins.kt

```diff
@@ -0,0 +1,300 @@
+            Message::class.java,
+            Long::class.javaObjectType,
+            MeUser::class.java,
+            Int::class.javaObjectType,
+            Boolean::class.javaPrimitiveType!!,
+            Boolean::class.javaPrimitiveType!!,
+            Boolean::class.javaPrimitiveType!!,
+        ) { (
+                param,
+                /* message */ _: Message,
+                permissions: Long?,
+                /* meUser */ _: MeUser,
+                /* guildMfaLevel */ _: Int,
+                isPrivateChannel: Boolean,
+                isSystemDM: Boolean,
+                isArchivedThread: Boolean,
+            ) ->
+            val result = param.result as ManageMessageContext
+
+            // TODO: After 2026/01/12, users will no longer be able to pin messages with the
+            // Manage Messages permission, and this line should be removed around the same time.
+            // https://discord.com/developers/docs/change-log#pin-permission-split
+            if (result.canTogglePinned) return@after
+
+            val isPrivateDM = isPrivateChannel && !isSystemDM
+            val isPinPermitted = isPrivateDM || PermissionUtils.can(PIN_MESSAGES_PERMISSION, permissions)
+            val canPin = isPinPermitted && !isArchivedThread
+            param.setResult(
+                ManageMessageContext(
+                    result.canManageMessages,
+                    result.canEdit,
+                    result.canDelete,
+                    result.canAddReactions,
+                    canPin,
+                    result.canMarkUnread
+                )
+            )
+        }
+    }
+
+    // Patches the permissions editor to show new permission
+    private fun patchPermissionsEditor() {
+        val pinMessagesCheckboxViewId = View.generateViewId()
+
+        // Patches the editor to add our new pin messages checkbox and change the information
+        // regarding the manage messages permission
+        patcher.before<WidgetChannelSettingsEditPermissions>(
+            "onViewBound",
+            View::class.java,
+        ) { (_, view: View) ->
+            val layoutId = Utils.getResId("channel_permissions_text_container", "id")
+            val layout: LinearLayout = view.findViewById(layoutId)
+            val manageMessagesCheckboxId = Utils.getResId("channel_permission_text_manage_messages", "id")
+            val manageMessagesCheckbox: TernaryCheckBox = view.findViewById(manageMessagesCheckboxId)
+            val index = layout.indexOfChild(manageMessagesCheckbox) + 1
+
+            // TODO: Should be changed after 2026/01/12, or when official client does
+            manageMessagesCheckbox.setLabel("Manage Messages ⚠")
+            manageMessagesCheckbox.setSubtext(
+                "Members with this permission can delete messages by other members or pin any message.*\n\n" +
+                "* Pinning messages now has a separate permission. This setting's behaviour will change soon."
+            )
```

Comment:
Could you use the setter instead of the method, or is it not generated for these fields? I'm not sure if it's generated or not